client/src/components/Onboarding/OnboardingAdmin.js

```diff
@@ -1,7 +1,291 @@
+          <div
+            key={uuidv4()}
+            styleName={`status ${
+              step.state === 'staffwait' ? 'staffwait' : ''
+            }`}
+          >
+            <OnboardingStatus step={step} />
+          </div>
+        ))}
+      </td>
+      <td>
+        {user.steps.map(step => (
+          <div
+            key={uuidv4()}
+            styleName={step.state === 'staffwait' ? 'staffwait' : ''}
+          >
+            {step.state === 'staffwait' && (
+              <OnboardingApproveActions
+                callback={action =>
+                  actionCallback(step.step, user.username, action)
+                }
+              />
+            )}
+          </div>
+        ))}
+      </td>
+      <td>
+        {user.steps.map(step => (
+          <div
+            key={uuidv4()}
+            styleName={step.state === 'staffwait' ? 'staffwait' : ''}
+          >
+            <OnboardingResetLinks
+              callback={action =>
+                actionCallback(step.step, user.username, action)
+              }
+            />
+          </div>
+        ))}
+      </td>
+      <td>
+        {user.steps.map(step => (
+          <div
+            key={uuidv4()}
+            styleName={step.state === 'staffwait' ? 'staffwait' : ''}
+          >
+            <Button
+              color="link"
+              styleName="action-link"
+              onClick={() => viewLogCallback(user, step)}
+            >
+              View Log
+            </Button>
+          </div>
+        ))}
+      </td>
+    </tr>
+  );
+};
+
+OnboardingAdminListUser.propTypes = {
+  user: onboardingUserPropType.isRequired,
+  viewLogCallback: PropTypes.func.isRequired
+};
+
+const OnboardingAdminList = ({ users, viewLogCallback }) => {
+  return (
+    <table styleName="users">
+      <thead>
+        <tr>
+          <th>User</th>
+          <th>Step</th>
+          <th>Status</th>
+          <th>Administrative Actions</th>
+          <th>&nbsp;</th>
+          <th>Log</th>
+        </tr>
+      </thead>
+      <tbody>
+        {users.map(user => (
+          <OnboardingAdminListUser
+            user={user}
+            key={user.username}
+            viewLogCallback={viewLogCallback}
+          />
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+OnboardingAdminList.propTypes = {
+  users: PropTypes.arrayOf(onboardingUserPropType).isRequired,
+  viewLogCallback: PropTypes.func.isRequired
+};
+
+const OnboardingAdmin = () => {
+  const dispatch = useDispatch();
+  const [eventLogModalParams, setEventLogModalParams] = useState(null);
+
+  const { users, offset, limit, total, query, loading, error } = useSelector(
+    state => state.onboarding.admin
+  );
+
+  const paginationCallback = useCallback(
+    page => {
+      dispatch({
+        type: 'FETCH_ONBOARDING_ADMIN_LIST',
+        payload: {
+          offset: (page - 1) * limit,
+          limit,
+          query
+        }
+      });
+    },
+    [offset, limit, query]
+  );
```

Comment:
Not required for this PR:
Are we paginating for visual or performance reasons? If visual, would it be more straightforward to paginate on the frontend? This would reduce delay

Reply:
We went back and forth on this one. There's a lot of performance penalties, and limiting queries to pages reduces them. Pagination on the backend reduces these performance penalties that are about 50% from serialization. For whatever reason, serialization in a JsonResponse is very slow - returning 100 user records takes almost 30 seconds. (This may be a Python thing, as I've discovered in other projects where serialization of large objects really impacted performance.)